File: packages/CorptieClientCore/Sources/CorptieClientCore/ClientCapabilities.swift
import Foundation

public struct ClientCapabilities: Decodable, Sendable {
    public struct Connection: Decodable, Sendable {
        public let mode: String
        public let deviceAuthentication: Bool
        public let remoteAccess: Bool
    }
    public let schemaVersion: Int
    public let service: String
    public let connection: Connection

    public var supportsRemoteConnection: Bool {
        schemaVersion == 1 && service == "corptie"
            && connection.mode == "authenticated-remote"
            && connection.deviceAuthentication && connection.remoteAccess
    }
}

File: apps/backend/src/application/clientCapabilities.mjs
// Public bootstrap contract. Never include paths, credentials, Provider config, or user data.
export function clientCapabilities() {
  return {
    schemaVersion: 1,
    service: "corptie",
    connection: { mode: "local-only", deviceAuthentication: false, remoteAccess: false },
    synchronization: { stateSnapshot: true, stateChanges: true, stateEvents: true, timelineWindow: true },
    resources: { portableFileAccess: false },
  };
}
